app/models/payment.rb

```diff
@@ -0,0 +1,16 @@
+class Payment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :allocations
+  has_many :expenses, through: :allocations
+
+  validates :name, presence: true
+  validates :amount, presence: true, numericality: { only_integer: true, greater_than: 0 }
```

Comment:
- [ ] Kindly be more specific in the `user` input validation, I figured out that if someone adds a very huge number then it is also accepting that as input, which is not a bad thing but unrealistic and makes your UI look very bad. Please make sure that your validation also covers edge cases.